Feat/623/route api

JasonFritsche opened the pull request:
#623

JasonFritsche committed:
feat: work in progress

JasonFritsche committed:
feat: migrate to filesystem route api

ghost commented:
<details>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=728&src=https%3A%2F%2Fgithub.com%2Fdistributeaid%2Fdistributeaid.org">Click on the image</a> for a new way to code review</b>
</summary>

  - Make big changes easier — review code in small groups of related files
  - Know where to start — see the whole change at a glance
  - Take a code tour — explore the change with an interactive tour
  - Make comments and review — all fully sync’ed with github

    [Try it now!](https://app.codesee.io/r/reviews?pr=728&src=https%3A%2F%2Fgithub.com%2Fdistributeaid%2Fdistributeaid.org)
</details>

[![Review these changes using an interactive CodeSee Map](https://s3.us-east-2.amazonaws.com/maps.codesee.io/images/github/distributeaid/distributeaid.org/728/af966163/38845e7fc5d799caf8aa96b9507be0c4c7c7ba4f.svg)](https://app.codesee.io/r/reviews?pr=728&src=https%3A%2F%2Fgithub.com%2Fdistributeaid%2Fdistributeaid.org)

#### Legend
![CodeSee Map Legend](https://app.codesee.io/Legend_v2.png)

JasonFritsche committed:
feat: create subregions page using filesystem route api

JasonFritsche commented:
@raae I think I've got the pages (routes, region, subregion) working (big thanks to @jtfairbank ) using the filesystem route api. Could you please take a look when you're able and let me know if I should change/refactor anything. In the meantime I will work on getting rid of the templates and migrating over the templates to the new pages. Thank you!

raae commented on src/pages/regions/index.tsx:
```diff
@@ -96,8 +97,11 @@ const RegionsPage: FC<Props> = ({
-export const pageQuery = graphql`
-  query RegionsQuery {
+export const query = graphql`
+  query MyQuery($id: String) {
```
suggestion: markdown query not in use, can be removed

JasonFritsche replied:
removed the markdown query

raae commented on src/pages/regions/index.tsx:
```diff
@@ -21,6 +21,7 @@ const RegionsPage: FC<Props> = ({
     regions: { nodes: regions },
   },
 }) => {
+  console.log('dataatatatata', regions)
```
nitpick: remove logging

JasonFritsche replied:
removed

raae commented on src/pages/regions/index.tsx:
```diff
@@ -21,6 +21,7 @@ const RegionsPage: FC<Props> = ({
   },
 }) => {
+  console.log('dataatatatata', regions)
   const createRegionHref = (region: Region) => {
```
suggestion: use [gatsbyPath](https://www.gatsbyjs.com/docs/reference/routing/file-system-route-api/#gatsbypath-example) approach in the query to making sure the you get correct slugified path

raae reviewed (commented):
Great work, added some [conventional comments](https://conventionalcomments.org/) below.

praise: {daSubregion.region__name} folder 🥳

suggestion(non-blocking): I would set a slug on regions and subregions similar to routes equal to the filename, see the route creation to see how to set them.

JasonFritsche committed:
feat: create slug for regions and subregions

jtfairbank commented:
Hey @JasonFritsche just a heads up that we've refactored our gatsby code to use typescript in #736.  Might need to rebase or merge those changes into here and adjust your changes to work with typescript.

JasonFritsche commented:
> Hey @JasonFritsche just a heads up that we've refactored our gatsby code to use typescript in #736. Might need to rebase or merge those changes into here and adjust your changes to work with typescript.

@jtfairbank I'll merge in the latest and see what happens. Currently trying to get this slug issue resolved. I don't have time to look at this today, but will be able to spend some time tomorrow and over the weekend.

JasonFritsche committed:
feat: remove region/route templates

JasonFritsche committed:
chore: merge in latest from saga branch

JasonFritsche committed:
feat: remove createPages calls from gatsby-node

JasonFritsche commented:
@raae @jtfairbank I've merged in the latest from `saga`. The e2e tests pass. I've removed the route and regions templates. I am however, unable to figure out how to create the slug for Region and Subregion. I've left my attempt at creating the slug in the draft PR.

coderbyheart commented on gatsby-node.js:
```diff
@@ -0,0 +1,61 @@
+const path = require('path')
```
That file should not be in here.

renovate[bot] committed:
fix(deps): update dependency gatsby to v4.24.4

jtfairbank commented:
Hey @JasonFritsche I can tackle the slugs today!

JasonFritsche commented:
> Hey @JasonFritsche I can tackle the slugs today!

@jtfairbank I really appreciate the help. This story is definitely one that's out of my comfort zone. I have availability this evening (after 1700 CST) if you'd be interested in pair programming.

raae commented:
I could join in some pairing after 19:30 tonight...

jtfairbank commented:
Hey @raae unfortunately you two are in very different but similar sounding timezones. :sweat_smile: CST is 7 hours behind CEST.

@JasonFritsche I'm still traveling so it'll be easier to tackle this independently unless you want to wait until the Code Jam next Monday (7-9pm CST)?  We can always review the code together after it's merged in to create a learning opportunity!

raae commented:
@JasonFritsche you want to pair at 20.00 CEST?

JasonFritsche commented:
@raae unfortunately I won't be available until 1700 CST which I believe is 0300 CEST. 

@jtfairbank I don't mind if you work on this feature. There are (I believe) two outstanding issues in this ticket, the slug creation/use, and removing the gatsby-node.js file. Both of these items are presenting a challenge to me.

jtfairbank committed:
fix: Fix a merge issue by adding gatsby Head back to routes page. Remove extraneous gatsby-node.js

jtfairbank committed:
feat: Use slugs with filesystemRouteAPI, fix broken image link.

raae commented:
Totally misunderstood the timezone issue 🤦‍♀️

jtfairbank committed:
test: Add basic e2e tests for regions / region / subregion pages.

jtfairbank committed:
refactor: use GatsbyPath to create region / subregion urls, not slugify.

jtfairbank committed:
Merge branch 'saga' into feat/623/route-api

jtfairbank commented:
Ok y'all here's what I've done:

  - Added e2e tests for the regions index page, region pages, and subregion pages.
  - Created slugs based on the filesystem for Regions / Subregions in `/gatsby/transform-nodes.ts`
  - Attempted to use GatsbyPath to get the links for the dynamic region & subregion pages.
  
Where I'm stuck:

  - The subregions use two parameters in their dynamic path which works with the fileSystemAPI but doesn't work with GatsbyPath.  It is attempting to create paths that look like this "/regions/{DASubregion.region__slug}/serbia/" instead of "/regions/the-balkans/serbia/"

@raae any insights here?

coderbyheart committed:
style: enable type checks for all TypeScript code

Note: some TypeScript errors have been silenced below,
to enable type checking for *new* code. Feel free to resolve.

renovate[bot] committed:
chore(deps): update dependency postcss to v8.4.18

raae commented:
I have not had that use case before @jtfairbank, so no insight 😬

renovate[bot] committed:
fix(deps): update playwright monorepo to v1.27.1

Commit:
Update from Forestry.io
Distribute Aid Admin created static/uploads/50297749_521554841683858_669757054109351936_n.jpg

Commit:
Update from Forestry.io
Distribute Aid Admin updated content/pages/our-story.md

Commit:
Update from Forestry.io
Distribute Aid Admin updated content/pages/our-story.md

Commit:
Update from Forestry.io - Updated Forestry configuration

Commit:
Update from Forestry.io
Distribute Aid Admin updated content/pages/our-story.md

Commit:
Update from Forestry.io
Distribute Aid Admin updated content/pages/our-story.md

Commit:
Update from Forestry.io - Updated Forestry configuration

Commit:
Update from Forestry.io - Updated Forestry configuration

Commit:
Update from Forestry.io - Updated Forestry configuration

Commit:
Update from Forestry.io - Updated Forestry configuration

Commit:
Update from Forestry.io
Distribute Aid Admin updated content/pages/our-story.md

Commit:
Update from Forestry.io
Distribute Aid Admin updated content/pages/our-story.md

Commit:
Update from Forestry.io
Distribute Aid Admin updated content/pages/our-story.md

Commit:
Update from Forestry.io
Distribute Aid Admin updated content/pages/our-story.md

Commit:
Update from Forestry.io
Distribute Aid Admin updated content/pages/our-story.md

renovate[bot] committed:
chore(deps): update dependency netlify-cli to v12.0.9

renovate[bot] committed:
chore(deps): update graphqlcodegenerator monorepo

jtfairbank committed:
feat: Setup consistent slugs n paths when region, subregions, and route nodes are created.

jtfairbank committed:
Merge branch 'saga' into feat/623/route-api

jtfairbank committed:
fix: fix typescript errors in transform-nodes

jtfairbank committed:
fix: fix typescript errors in transform-nodes

coderbyheart commented on gatsby-node.ts:
```diff
@@ -42,9 +41,7 @@ Create Dynamic Pages
-  await pages.createRegionPages(args)
-  await pages.createSubregionPages(args)
-  await pages.createRoutePages(args)
+  // Currently all pages are created using the Filesystem Route API
```
Remove entire block, because `createPages` is not used.

coderbyheart committed:
style: remove unused code

coderbyheart committed:
style: organize imports

coderbyheart committed:
style: get rid of @ts-ignore and handle errors

coderbyheart committed:
fix: use exact version

coderbyheart reviewed (approved):
Feedback in #768

jtfairbank commented:
Hey @coderbyheart! #768 looks great, we can merge it into here once it passes CI, then merge this into saga?

coderbyheart commented:
Will do!

coderbyheart committed:
Merge branch 'feat/623/route-api-review-coderbyheart' into feat/623/route-api

coderbyheart commented:
Thanks everyone! 🚀

jtfairbank commented:
Thank you for carrying this one home @coderbyheart!
